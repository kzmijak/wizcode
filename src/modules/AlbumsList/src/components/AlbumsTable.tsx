import { Table } from "@mui/joy";
import { AlbumModel } from "models/Album";
import { FC } from "react";

type AlbumsTableProps = {
  rows: AlbumModel[];
};

export const AlbumsTable: FC<AlbumsTableProps> = ({ rows }) => {
  return (
    <Table
      aria-label="albums table"
      borderAxis="none"
      color="neutral"
      size="md"
      stickyHeader
      stripe="odd"
      variant="plain"
    >
      <thead>
        <tr>
          <th>#</th>
          <th>Image</th>
          <th>Title</th>
          <th>Artist</th>
        </tr>
      </thead>
      <tbody>
        {rows.map((row, index) => (
          <tr key={row.id}>
            <td>{index + 1}</td>
            <td>
              <img src={row.imageUrl} alt={`${row.title} album image`} />
            </td>
            <td>{row.title}</td>
            <td>{row.artistName}</td>
          </tr>
        ))}
      </tbody>
    </Table>
  );
};
