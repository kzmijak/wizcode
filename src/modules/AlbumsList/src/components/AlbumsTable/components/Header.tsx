import { FC } from "react";
import { Search } from "./utils/Search";

type HeaderProps = {
  search: string;
  onSearchChange: (value: string) => void;
};

export const Header: FC<HeaderProps> = ({ onSearchChange, search }) => {
  return (
    <section>
      <Search value={search} onChange={onSearchChange} />
    </section>
  );
};
