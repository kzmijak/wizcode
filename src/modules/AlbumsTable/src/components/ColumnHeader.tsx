import { Box, Typography } from "@mui/joy";
import { FC } from "react";
import { columnWidths } from "../config/columnWidths";
import { AlbumsTableColumn } from "../models/AlbumsTableColumn";
import { columnHeaderNames } from "../config/columnHeaderNames";

type ColumnHeaderProps = {
  column: AlbumsTableColumn;
};

export const ColumnHeader: FC<ColumnHeaderProps> = ({ column }) => {
  return (
    <Box component="th" width={columnWidths[column]}>
      <Typography>{columnHeaderNames[column]}</Typography>
    </Box>
  );
};
