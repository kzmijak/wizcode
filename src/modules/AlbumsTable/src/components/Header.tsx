import { FC } from "react";
import { Search } from "./utils/Search";

type TableHeaderProps = {
  search: string;
  onSearchChange: (value: string) => void;
};

export const TableHeader: FC<TableHeaderProps> = ({
  onSearchChange,
  search,
}) => {
  return (
    <section>
      <Search value={search} onChange={onSearchChange} />
    </section>
  );
};
