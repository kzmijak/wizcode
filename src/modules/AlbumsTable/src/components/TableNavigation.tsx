import { FC } from "react";
import { Search } from "./filters/Search";

type TableHeaderProps = {
  search: string;
  onSearchChange: (value: string) => void;
};

export const TableNavigation: FC<TableHeaderProps> = ({
  onSearchChange,
  search,
}) => {
  return (
    <section>
      <Search value={search} onChange={onSearchChange} />
    </section>
  );
};
