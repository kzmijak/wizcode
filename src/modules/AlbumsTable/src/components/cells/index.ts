export { IndexCell } from "./IndexCell";
export { ImageCell } from "./ImageCell";
export { TitleCell } from "./TitleCell";
export { ArtistCell } from "./ArtistCell";
