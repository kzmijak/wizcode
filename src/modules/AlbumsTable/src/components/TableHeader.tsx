import { FC } from "react";
import { albumsTableColumnConsts } from "../models/AlbumsTableColumn";
import { ColumnHeader } from "./ColumnHeader";

export const TableHeader: FC = () => {
  return (
    <thead>
      <tr>
        {albumsTableColumnConsts.map((column) => (
          <ColumnHeader column={column} />
        ))}
      </tr>
    </thead>
  );
};
