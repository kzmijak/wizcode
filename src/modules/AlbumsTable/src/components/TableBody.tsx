import { FC } from "react";
import { AlbumsTableRow } from "../models/AlbumsTableRow";
import { IndexCell, ImageCell, TitleCell, ArtistCell } from "./cells";

type TableBodyProps = {
  allRows: AlbumsTableRow[];
  displayedRows: AlbumsTableRow[];
};

export const TableBody: FC<TableBodyProps> = ({ allRows, displayedRows }) => {
  return (
    <tbody>
      {displayedRows.map((row) => (
        <tr key={row.id}>
          <IndexCell currentRow={row} rows={allRows} />
          <ImageCell currentRow={row} />
          <TitleCell currentRow={row} />
          <ArtistCell currentRow={row} />
        </tr>
      ))}
    </tbody>
  );
};
