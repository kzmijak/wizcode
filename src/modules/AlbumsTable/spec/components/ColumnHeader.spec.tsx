import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { ColumnHeader } from "../../src/components/ColumnHeader";

describe("ColumnHeader", () => {
  const Component = () => <ColumnHeader column="Artist" />;

  it("should match snapshot", () => {
    expect(render(<Component />)).toMatchSnapshot();
  });
});
