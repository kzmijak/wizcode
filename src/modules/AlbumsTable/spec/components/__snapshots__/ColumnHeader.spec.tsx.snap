// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ColumnHeader > should match snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <th
        class="MuiBox-root css-j3puv3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
        >
          Artist
        </p>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="MuiBox-root css-j3puv3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
      >
        Artist
      </p>
    </th>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColumnHeader > should match snapshot for Artist 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <th
        class="MuiBox-root css-j3puv3"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
        >
          Artist
        </p>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="MuiBox-root css-j3puv3"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
      >
        Artist
      </p>
    </th>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColumnHeader > should match snapshot for Image 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <th
        class="MuiBox-root css-1bvg7gj"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
        >
          Image
        </p>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="MuiBox-root css-1bvg7gj"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
      >
        Image
      </p>
    </th>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColumnHeader > should match snapshot for Index 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <th
        class="MuiBox-root css-1x769y4"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
        >
          #
        </p>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="MuiBox-root css-1x769y4"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
      >
        #
      </p>
    </th>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`ColumnHeader > should match snapshot for Title 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <th
        class="MuiBox-root css-crm8vv"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
        >
          Title
        </p>
      </th>
    </div>
  </body>,
  "container": <div>
    <th
      class="MuiBox-root css-crm8vv"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-lmmucf-JoyTypography-root"
      >
        Title
      </p>
    </th>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
